schema/group.py

```diff
@@ -0,0 +1,50 @@
+import re
+
+from marshmallow import Schema, fields, validates, ValidationError
+
+
+class GroupNameSchemaBase(Schema):
+    name = fields.Str(required=True, allow_none=False)
+
+    @validates("name")
+    def validate_name(self, name):
+        if not re.match("^[A-Za-z0-9_-]{1,32}$", name):
+            raise ValidationError(
+                "Group should contain max 32 chars and include only letters, digits, underscores and dashes"
+            )
+
+
+class GroupCreateRequestSchema(Schema):
+    capabilities = fields.List(fields.Str(), missing=[])
+
+
+class GroupUpdateRequestSchema(Schema):
+    name = fields.Str(missing=None)
+    capabilities = fields.List(fields.Str(), missing=None)
+
+    @validates("name")
+    def validate_name(self, name):
+        if name is not None and not re.match("^[A-Za-z0-9_-]{1,32}$", name):
```

Comment:
This is a weird combo, it should be investigated how `missing=None` interacts with validator with `is not None`.
Consider using GroupNameSchemaBase.